src/binding/python/RecordComponent.cpp

```diff
@@ -108,9 +108,92 @@ void init_RecordComponent(py::module &m) {
 
         // deprecated: pass-through C++ API
         .def("load_chunk", [](RecordComponent & r, Offset const & offset, Extent const & extent) {
-            std::vector<ptrdiff_t> c(extent.size());
-            std::copy(std::begin(extent), std::end(extent), std::begin(c));
-            return py::array( c, r.loadChunk<double>(offset, extent).get() );
+            std::vector<ptrdiff_t> shape(extent.size());
+            std::copy(std::begin(extent), std::end(extent), std::begin(shape));
+
+            auto dtype = py::dtype("void");
+
+            if( r.getDatatype() == Datatype::CHAR ) dtype = py::dtype("b");
+            else if( r.getDatatype() == Datatype::UCHAR ) dtype = py::dtype("B");
+            else if( r.getDatatype() == Datatype::INT16 ) dtype = py::dtype("int16");
+            else if( r.getDatatype() == Datatype::INT32 ) dtype = py::dtype("int32");
+            else if( r.getDatatype() == Datatype::INT64 ) dtype = py::dtype("int64");
+            else if( r.getDatatype() == Datatype::UINT16 ) dtype = py::dtype("uint16");
+            else if( r.getDatatype() == Datatype::UINT32 ) dtype = py::dtype("uint32");
+            else if( r.getDatatype() == Datatype::UINT64 ) dtype = py::dtype("uint64");
+            else if( r.getDatatype() == Datatype::LONG_DOUBLE ) dtype = py::dtype("longdouble");
+            else if( r.getDatatype() == Datatype::DOUBLE ) dtype = py::dtype("double");
+            else if( r.getDatatype() == Datatype::FLOAT ) dtype = py::dtype("float");
+
+            /* @todo
```

Comment:
All datatypes in this block are (currently) invalid arguments for chunk transfers.